src/System.Runtime/tests/System/StringGetHashCodeTests.cs

```diff
@@ -0,0 +1,66 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using Xunit;
+
+namespace System.Tests
+{
+    public class StringGetHashCodeTests : RemoteExecutorTestBase
+    {
+        /// <summary>
+        /// Ensure that hash codes are randomized by getting the hash in two processes
+        /// and confirming it is different (modulo possible values of int).
+        /// If the legacy hash codes are being returned, it will not be different.
+        /// </summary>
+        [Theory]
+        [MemberData(nameof(GetHashCode_TestData))]
+        public void GetHashCodeWithStringComparer_UseSameStringInTwoProcesses_ReturnsDifferentHashCodes(int getHashCodeIndex)
+        {
+            Func<string, string, int> method = (parentHash, i) => { return int.Parse(parentHash) != s_GetHashCodes[int.Parse(i)]() ? 0 : -1; };
+            int parentHashCode = s_GetHashCodes[getHashCodeIndex]();
+            int exitCode, timesTried = 0;
+            do
+            {
+                // very small chance the child and parent hashcode are the same. To further reduce chance of collision we try up to 3 times
+                using (RemoteInvokeHandle handle = RemoteInvoke(method, parentHashCode.ToString(), getHashCodeIndex.ToString(), new RemoteInvokeOptions { CheckExitCode = false }))
```

Comment:
I could only pass down string variables as arguments to RemoteInvoke.
This is why I pass down the index and parenthashcodes as string
I did `int.Parse` rather than `TryParse` because I'm sure I'm passing down ints.